extlib/scheduler-playground/lib/store.js

```diff
@@ -0,0 +1,38 @@
+var _ = require('underscore');
+
+var Store = function(options) {
+    // First and last points in the store
+    this.last  = new Date(options.last.getTime());
+    this.first = new Date(options.first.getTime());
+
+    // Points per second (data density)
+    this.density = options.density;
+
+    this.data = [];
+
+    this.init();
+};
+
+Store.prototype.init = function() {
+    var range = this.last.getTime() - this.first.getTime();
+    var maxPoints = (range / 1000) * this.density;
+
+    while (this.data.length < maxPoints) {
```

Comment:
What happens now is, that the number of points will always be `maxPoints` in the time interval - the average will equal to `density`, but the times are generated randomly (from a uniform distribution between `first` and `last`. That they would all fall into a sub-interval is possible, but won't happen on the average.

In that case, the adapter would just crunch the points in the first quarter as fast as it's queue size allows, then process the remaining few (or zero) points by querying with `from/to = (1/4, 1)`

Does that answer your question? I think that the comment above the `density` option is misleading and I should probably change that.

Also, for debugging purposes, to be able to specify a more regular sequence (e.g. tick every nth of a second) would be useful.